libsolidity/ast/AST.h

```diff
@@ -1521,7 +1530,7 @@ class WhileStatement: public BreakableStatement
 /**
  * For loop statement
  */
-class ForStatement: public BreakableStatement, public Scopable
+class ForStatement: public BreakableStatement, public Scopable, public ScopeOpener
```

Comment:
Hm, the while statement does not create its own scope?

Reply:
No, its block does. For the for loop, we need a new scope because of the init part which is not a block.